src/ctim/domain/sorting.cljc

```diff
@@ -15,22 +16,30 @@
   depends on a stable 'now', so it must be passed in.  Returns -1, 0,
   or 1, as per 'compare."
   [dt-now
-   {{j1_start :start_time} :valid_time
+   {{j1_start :start_time
+     j1_end   :end_time} :valid_time
     j1_disp :disposition
-    :as j1}
-   {{j2_start :start_time} :valid_time
+    :as _judgement-1_}
+   {{j2_start :start_time
+     j2_end   :end_time} :valid_time
     j2_disp :disposition
-    :as j2}]
-  (let [;; Validity is reverse sorted so that true comes before false
-        result1 (compare (validity/valid-now? dt-now j2)
-                         (validity/valid-now? dt-now j1))]
+    :as _judgement-2_}]
+  (let [j1_start (time-coerce/to-internal-date j1_start)
+        j1_end   (time-coerce/to-internal-date j1_end)
+        j2_start (time-coerce/to-internal-date j2_start)
+        j2_end   (time-coerce/to-internal-date j2_end)
+
+        ;; Validity is reverse sorted so that true comes before false
+        result1 (compare (validity/valid-now? dt-now j2_start j2_end)
+                         (validity/valid-now? dt-now j1_start j1_end))]
     (if (not= 0 result1)
       result1
       (let [result2 (compare j1_disp j2_disp)]
         (if (not= 0 result2)
           result2
           ;; Start-time is reverse sorted
-          (compare j2_start j1_start))))))
+          (compare (time-coerce/to-long j2_start)
+                   (time-coerce/to-long j1_start)))))))
```

Comment:
The compare function is a bit hard to follow for me.
Would it be possible to rename some temporary values with more meaningful names?

Instead of `result1`, `result2` why not `valid-time-comparison`, `disposition-comparison`?